dbms/src/Flash/Coprocessor/DAGContext.h

```diff
@@ -406,6 +408,10 @@ class DAGContext
     // In disaggregated tiflash mode, table_scan in tiflash_compute node will be converted ExchangeReceiver.
     // Record here so we can add to receiver_set and cancel/close it.
     std::optional<std::pair<String, ExchangeReceiverPtr>> disaggregated_compute_exchange_receiver;
+
+    /// Holding the table lock to make sure that the table wouldn't be dropped during the lifetime of this query,
+    /// even if there are no local regions.
+    TableLockHolders table_locks;
```

Comment:
After this pr, the locks may be released before the data streams, before this pr, the locks will be released when data streams are destructed, is it the expected behavior?

Reply:
There may be a problem with cancel because the asynchronous table scan will not stop until after the UnorderedInputStream is destroyed...

Reply:
fixed.